kernel/kernel-api/src/main/java/io/delta/kernel/internal/DeltaLogActionUtils.java

```diff
@@ -0,0 +1,286 @@
+                                  LongType.LONG),
+                      "Get the expression evaluator for the commit version");
+              ExpressionEvaluator commitTimestampGenerator =
+                  wrapEngineException(
+                      () ->
+                          engine
+                              .getExpressionHandler()
+                              .getEvaluator(readSchema, Literal.ofLong(timestamp), LongType.LONG),
+                      "Get the expression evaluator for the commit timestamp");
+              ColumnVector commitVersionVector =
+                  wrapEngineException(
+                      () -> commitVersionGenerator.eval(actionWrapper.getColumnarBatch()),
+                      "Evaluating the commit version expression");
+              ColumnVector commitTimestampVector =
+                  wrapEngineException(
+                      () -> commitTimestampGenerator.eval(actionWrapper.getColumnarBatch()),
+                      "Evaluating the commit timestamp expression");
+
+              return actionWrapper
+                  .getColumnarBatch()
+                  .withNewColumn(0, COMMIT_VERSION_STRUCT_FIELD, commitVersionVector)
+                  .withNewColumn(1, COMMIT_TIMESTAMP_STRUCT_FIELD, commitTimestampVector);
+            });
+  }
+
+  //////////////////////
+  // Private helpers //
+  /////////////////////
+
+  /** Column name storing the commit version for a given file action */
+  private static final String COMMIT_VERSION_COL_NAME = "version";
+
+  private static final DataType COMMIT_VERSION_DATA_TYPE = LongType.LONG;
+  private static final StructField COMMIT_VERSION_STRUCT_FIELD =
+      new StructField(COMMIT_VERSION_COL_NAME, COMMIT_VERSION_DATA_TYPE, false /* nullable */);
+
+  /** Column name storing the commit timestamp for a given file action */
+  private static final String COMMIT_TIMESTAMP_COL_NAME = "timestamp";
+
+  private static final DataType COMMIT_TIMESTAMP_DATA_TYPE = LongType.LONG;
+  private static final StructField COMMIT_TIMESTAMP_STRUCT_FIELD =
+      new StructField(COMMIT_TIMESTAMP_COL_NAME, COMMIT_TIMESTAMP_DATA_TYPE, false /* nullable */);
+
+  /**
+   * Given a list of delta versions, verifies that they are (1) continuous (2) versions starts with
+   * expectedStartVersion and (3) end with expectedEndVersion. Throws an exception if any of these
+   * are not true.
+   *
+   * <p>Public to expose for testing only.
+   *
+   * @param commitFiles in sorted increasing order according to the commit version
+   */
+  public static void verifyDeltaVersions(
+      List<FileStatus> commitFiles,
+      long expectedStartVersion,
+      long expectedEndVersion,
+      Path tablePath) {
+
+    List<Long> commitVersions =
+        commitFiles.stream()
+            .map(fs -> FileNames.deltaVersion(new Path(fs.getPath())))
+            .collect(Collectors.toList());
+
+    if (!commitVersions.isEmpty()) {
+      List<Long> contVersions =
```

Comment:
a for loop would be more efficient btw, and easier to reason about

```java
for (int i = 1; i < commitVersions.size(); i++) {
      if (commitVersions.get(i) != commitVersions.get(i - 1) + 1) {
        throw new InvalidTableException(
            tablePath.toString(),
            String.format("Missing delta files: versions are not contiguous: (%s)", commitVersions));
      }
    }
```

Reply:
Sure, I mostly just copied this function from the log segment code